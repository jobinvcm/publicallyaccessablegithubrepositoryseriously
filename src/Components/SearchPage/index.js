import React from "react";
import { withStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import Paper from "@material-ui/core/Paper";
import InputBase from "@material-ui/core/InputBase";
import IconButton from "@material-ui/core/IconButton";
import propTypes from "prop-types";

const styles = theme => ({
  container: {
    width: "300px",
    position: "absolute",
    zIndex: "10",
    left: "50%",
    transform: "translateX(-50%)",
    bottom: "44px",
    overflow: "hidden"
  },
  root: {
    backgroundColor: "#fff",
    borderRadius: "25px",
    width: "100%"
  },
  inputBase: {
    display: "block",
    borderRadius: "5px",
    padding: "4px 16px",
  },
  iconButton: {
    color: "#01D277",
    position: "absolute",
    right: 0,
    bottom: "50%",
    transform: "translateY(50%)"
  },
  paperContainer: {
    borderRadius: "16px"
  }
});

const SearchPage = props => {
  const { classes, onChange } = props;
  return (
    <div className={classes.container}>
      <Paper elevation={20} className={classes.paperContainer}>
        <InputBase className={classes.inputBase} onChange={onChange} placeholder="Search Movies" />
        <IconButton
          className={classes.iconButton}
          aria-label="Search"
        >
          <SearchIcon />
        </IconButton>
      </Paper>
    </div>
  );
};

SearchPage.propTypes = {
  classes: propTypes.object,
}

export default withStyles(styles)(SearchPage);
