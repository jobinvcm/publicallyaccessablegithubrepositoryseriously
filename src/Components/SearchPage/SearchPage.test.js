import React from 'react';
import { shallow } from 'enzyme';
import SearchPage from '.';
import Paper from '@material-ui/core/Paper';
import SearchIcon from "@material-ui/icons/Search";
import IconButton from "@material-ui/core/IconButton";
import InputBase from "@material-ui/core/InputBase";

describe('<SearchPage />', () => {
  it('renders paper component', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should should have input', () => {
    const mock = jest.fn();
    const wrapper = shallow(<SearchPage onChange={mock} />);
    expect(wrapper.dive().find(InputBase)).toHaveLength(1);
  });

  it('should should have IconButton', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper.dive().find(Paper).find(IconButton)).toHaveLength(1);
  });

  it('should should have Search Icon', () => {
    const wrapper = shallow(<SearchPage />);
    expect(wrapper.dive().find(Paper).find(IconButton).find(SearchIcon)).toHaveLength(1);
  });

  it('should trigger onchange function', () => {
    const mock = jest.fn();
    const wrapper = shallow(<SearchPage onChange={mock} />);
    const input = wrapper.dive().find(InputBase);
    input.simulate('change', {target: { value: 'changed'}});
    expect(mock).toHaveBeenCalled();
  });  
})
