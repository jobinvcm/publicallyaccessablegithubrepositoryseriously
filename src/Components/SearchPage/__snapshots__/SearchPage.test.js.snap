// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> renders paper component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SearchPage) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container": "SearchPage-container-1",
        "iconButton": "SearchPage-iconButton-4",
        "inputBase": "SearchPage-inputBase-3",
        "paperContainer": "SearchPage-paperContainer-5",
        "root": "SearchPage-root-2",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container": "SearchPage-container-1",
          "iconButton": "SearchPage-iconButton-4",
          "inputBase": "SearchPage-inputBase-3",
          "paperContainer": "SearchPage-paperContainer-5",
          "root": "SearchPage-root-2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
